dashboard/app/models/script.rb

```diff
@@ -199,13 +199,16 @@ def self.stage_extras_script_ids
   # @param [User] user
   # @return [Script[]]
   def self.valid_scripts(user)
-    with_hidden = user.permission?(UserPermission::HIDDEN_SCRIPT_ACCESS)
+    user_experiments_enabled = Course.has_any_course_experiments?(user)
+    with_hidden = !user_experiments_enabled && user.hidden_script_access?
     cache_key = "valid_scripts/#{with_hidden ? 'all' : 'valid'}"
-    Rails.cache.fetch(cache_key) do
+    scripts = Rails.cache.fetch(cache_key) do
```

Comment:
once we have i18n working here, the locale will need to be part of the cache key, similar to https://github.com/code-dot-org/code-dot-org/blob/c13a61f63a6e344767ce42c09c7dbe0874d60b65/dashboard/app/models/course.rb#L228